components/micro-gateway-core/src/main/ballerina/src/gateway/handlers/basic_auth_handler.bal

```diff
@@ -0,0 +1,76 @@
+// Copyright (c) 2020, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+//
+// WSO2 Inc. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import ballerina/auth;
+import ballerina/http;
+import ballerina/runtime;
+import ballerina/stringutils;
+
+# Representation of the basic auth handler
+#
+# + basicAuthProvider - The reference to the basic auth provider instance
+public type BasicAuthHandler object {
+
+    *http:InboundAuthHandler;
+
+    public BasicAuthProvider basicAuthProvider;
+
+    public function __init(BasicAuthProvider basicAuthProvider) {
+        self.basicAuthProvider = basicAuthProvider;
+    }
+
+    # Checks if the request can be authenticated with the Basic Auth header.
+    #
+    # + req - The `Request` instance.
+    # + return - Returns `true` if can be authenticated. Else, returns `false`.
+    public function canProcess(http:Request req) returns @tainted boolean {
+        string authHeader = runtime:getInvocationContext().attributes[AUTH_HEADER].toString().trim();
```

Comment:
There is a method to get authorization header from the service annotation.  Is that method setting the value to the context

Reply:
In the pre authentication filter, it is added to the invocation context. Besides that, it is required to pass the filter context to call that specific method.